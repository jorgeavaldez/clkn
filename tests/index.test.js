const main = require('../src/time');

describe('index.test.js', async () => {
  describe('parseTime', async () => {
    it('should accept a time and return a moment time', async () => {
      const testTime = '330pm';

      const result = main.parseTime(testTime);
      expect(result.format()).toMatchSnapshot();
    });
  });

  describe('timeDiff', async () => {
    it('should accept two times and return the difference', async () => {
      let startTime = '330pm';
      let endTime = '430pm';

      expect(main.timeDiff(startTime, endTime)).toEqual(1);

      endTime = '630pm';
      expect(main.timeDiff(startTime, endTime)).toEqual(3);

      endTime = '230pm';
      expect(main.timeDiff(startTime, endTime)).toEqual(-1);

      endTime = '200pm';
      expect(main.timeDiff(startTime, endTime)).toEqual(-1.5);
    });
  });
});
