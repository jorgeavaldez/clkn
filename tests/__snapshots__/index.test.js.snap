// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.test.js parseTime should accept a time and return a moment time 1`] = `"2018-02-14T15:30:00-06:00"`;
